src/main/java/de/tum/in/www1/artemis/service/ProgrammingSubmissionService.java

```diff
@@ -577,22 +579,25 @@ public ProgrammingSubmission findByResultId(long resultId) throws EntityNotFound
      *
      * @param exerciseId - the id of the exercise we are looking for
      * @param correctionRound - the correctionRound for which the submissions should be fetched for
-     * @param tutorId    - the id of the tutor we are interested in
+     * @param tutor    - the the tutor we are interested in
      * @param examMode - flag should be set to ignore the test run submissions
      * @return a list of programming submissions
      */
-    public List<ProgrammingSubmission> getAllProgrammingSubmissionsAssessedByTutorForCorrectionRoundAndExercise(long exerciseId, long tutorId, boolean examMode,
+    public List<ProgrammingSubmission> getAllProgrammingSubmissionsAssessedByTutorForCorrectionRoundAndExercise(long exerciseId, User tutor, boolean examMode,
             Long correctionRound) {
-        List<StudentParticipation> participations;
+        List<Submission> submissions;
         if (examMode) {
-            participations = this.studentParticipationRepository.findWithLatestSubmissionByExerciseAndAssessorAndCorrectionRoundIgnoreTestRuns(exerciseId, tutorId,
-                    correctionRound);
+            var participations = this.studentParticipationRepository.findAllByParticipationExerciseIdAndResultAssessorAndCorrectionRoundIgnoreTestRuns(exerciseId, tutor);
+            submissions = participations.stream().map(StudentParticipation::findLatestSubmission).filter(Optional::isPresent).map(Optional::get).map(submission -> {
+                submission.setResults(submission.getManualAndNullResults());
+                return submission;// .getResults().size() - 1 >= correctionRound && submission.getResults().get(correctionRound.intValue()) != null;
```

Comment:
This looks like dead code which should probably be removed?

Reply:
Yes you are right :) I removed it